src/main/java/com/github/mkopylec/sessioncouchbase/persistent/CouchbaseSession.java

```diff
@@ -60,12 +60,12 @@ public static String globalAttributeName(String attributeName) {
 
     @Override
     public long getCreationTime() {
-        return round((double) globalAttributes.get(CREATION_TIME_ATTRIBUTE));
+        return round(new Double(globalAttributes.get(CREATION_TIME_ATTRIBUTE).toString()));
```

Comment:
What version of Couchbase are you using?
For developing the project I use 4.0.0 Community edition. This version stores every number as decimal, so when reading the session from Couchbase every number is a double.
I'm curious how you manage to produce the above situation. Could you please provide a small sample project that I can clone and check it out?